documentation/modules/exploit/unix/http/cacti_filter_sqli_rce.md

```diff
@@ -0,0 +1,83 @@
+## Vulnerable Application
+
+This module exploits a SQL injection vulnerability in Cacti 1.2.12 and before.
+An admin can exploit the filter variable within `color.php` to pull arbitrary
+values as well as conduct stacked queries. With stacked queries, the
+`path_php_binary` value is changed within the settings table to a payload,
+and an update is called to  execute the payload.
+
+After calling the payload, the value is reset.  This has to be done from
+a new login session as any requests with the already logged in cookie
+will timeout.
+
+### Install
+
+As of the writing of this module, Ubuntu's 12.04 repo had cacti as
+1.2.10, so a simple `apt-get install cacti` will install a vulnerable
+version.
+
```

Comment:
My steps to get a specific version with docker

```shell
git clone https://github.com/scline/docker-cacti.git
cd docker-cacti

rm -rf cacti
git checkout 1.2.11 -- cacti
cp ./docker-compose/cacti_testing.yml ./docker-compose.yml
sudo docker-compose up
```

Navigate to `localhost:80` and login with `admin:admin` and run through the install steps.

Reply:
I also installed nc to make life easier with the default payload options:

```
sudo docker-compose exec cacti /bin/bash
yum install nc
```